crates/ruff_python_formatter/src/string/any.rs

```diff
@@ -54,23 +55,21 @@ impl<'a> AnyString<'a> {
         }
     }
 
-    /// Returns a vector of all the [`AnyStringPart`] of this string.
-    pub(super) fn parts(self, quoting: Quoting) -> AnyStringPartsIter<'a> {
+    /// Returns an iterator over the [`AnyStringPart`]s of this string.
+    pub(super) fn parts(self) -> AnyStringPartsIter<'a> {
         match self {
             Self::String(ExprStringLiteral { value, .. }) => {
                 AnyStringPartsIter::String(value.iter())
             }
             Self::Bytes(ExprBytesLiteral { value, .. }) => AnyStringPartsIter::Bytes(value.iter()),
-            Self::FString(ExprFString { value, .. }) => {
-                AnyStringPartsIter::FString(value.iter(), quoting)
-            }
+            Self::FString(ExprFString { value, .. }) => AnyStringPartsIter::FString(value.iter()),
         }
     }
 
     pub(crate) fn is_multiline(self, source: &str) -> bool {
         match self {
             AnyString::String(_) | AnyString::Bytes(_) => {
-                self.parts(Quoting::default())
```

Comment:
This is kind of what triggered the refactor. It felt off that it is necessary to pass an arbitrary `Quoating` to iterate over the parts.